### PR TITLE
feat(pci.ai.apps): datatr-80 - add scaling option

### PR DESCRIPTION
**Pull request by jperchoc**
DATATR-80

| Question         | Answer
| ---------------- | ---
| Branch?          | `release/ai-deploy-beta`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #DATATR-80
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Add scaling option to ai deploy apps

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/ai/apps/components/update-scaling/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "pci_app_update_app_scaling_title": "Modification du dimensionnement",
+  "pci_app_update_app_scaling_submit_label": "Confirmer",
+  "pci_app_update_app_scaling_cancel_label": "Annuler",
+  "pci_app_update_app_scaling_description": "Selon votre préférence, choisissez la stratégie de déploiement qui convient le mieux pour votre application : optez pour un dimensionnement fixe du nombre de réplicas sur lesquelles sera déployée votre application ou bien indiquez un minimum et maximum de réplicas pour vos déploiements afin d'en optimiser dynamiquement les performances et vous adapter au nombre de requêtes via l'option Autoscaling.",
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/ai/apps/components/update-scaling/translations/Messages_fr_FR.json:5]: i noticed we use `app` in some translation and `application` here.
May be we could use `app`.

File: packages/manager/modules/pci/src/projects/project/ai/apps/add/components/app-resources/app-resources.controller.js [hunk 2]
```diff
@@ -60,4 +69,32 @@ export default class AppResourcesController {
...
+  get tax() {
+    const replicas = this.appModel.scalingStrategy.autoscaling
+      ? this.appModel.scalingStrategy.automatic.replicasMin
+      : this.appModel.scalingStrategy.fixed.replicas;
+    return (
+      (this.resourcePriceTax + this.partnerPriceTax) *
+      this.appModel.resource.nbResources *
+      replicas
+    );
+  }
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/ai/apps/add/components/app-resources/app-resources.controller.js:99]: **nitpick:** Duplicated logic can be avoided.

File: packages/manager/modules/pci/src/projects/project/ai/apps/dashboard/general-information/update-scaling/update-scaling.routing.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state(
+    'pci.projects.project.ai.apps.dashboard.general-information.update-scaling',
+    {
+      url: '/update-scaling',
+      // params: { app: null },
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/ai/apps/dashboard/general-information/update-scaling/update-scaling.routing.js:6]: **nitpick:** Can you remove the comment if params are not needed?

File: packages/manager/modules/pci/src/projects/project/ai/apps/dashboard/general-information/update-scaling/update-scaling.routing.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      views: {
+        modal: {
+          component: 'ovhManagerPciProjectAppsUpdateAppScaling',
+        },
+      },
+      layout: 'modal',
+      resolve: {
+        // app: /* @ngInject */ ($transition$) => $transition$.params().app,
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/ai/apps/dashboard/general-information/update-scaling/update-scaling.routing.js:14]: **nitpick:** Can you remove the comment if `app` is not required?

File: packages/manager/modules/pci/src/projects/project/ai/apps/components/update-scaling/translations/Messages_fr_FR.json (hunk 1 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/ai/apps/components/update-scaling/translations/Messages_fr_FR.json]: Good point, I've changed the wording here: https://github.com/ovh/manager/pull/8001/commits/33a7214d2b0990fa80eb54768cc5b204824233b1

File: packages/manager/modules/pci/src/projects/project/ai/apps/dashboard/general-information/update-scaling/update-scaling.routing.js (hunk 4 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/ai/apps/dashboard/general-information/update-scaling/update-scaling.routing.js]: Sure ! https://github.com/ovh/manager/pull/8001/commits/33a7214d2b0990fa80eb54768cc5b204824233b1

File: packages/manager/modules/pci/src/projects/project/ai/apps/dashboard/general-information/update-scaling/update-scaling.routing.js (hunk 3 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/ai/apps/dashboard/general-information/update-scaling/update-scaling.routing.js]: https://github.com/ovh/manager/pull/8001/commits/33a7214d2b0990fa80eb54768cc5b204824233b1

**Review by mohammed-zahaf — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/ai/apps/add/add.controller.js [hunk 5]
```diff
@@ -87,6 +98,11 @@ export default class AppAddController {
     });
   }
 
+  static buildScalingeBody(scalingStrategy) {
```
Comment (Dimfacion) [inline packages/manager/modules/pci/src/projects/project/ai/apps/add/add.controller.js:101]:
```suggestion
  static buildScalingBody(scalingStrategy) {
```

File: packages/manager/modules/pci/src/projects/project/ai/apps/add/add.controller.js [hunk 6]
```diff
@@ -112,11 +128,7 @@ export default class AppAddController {
       partnerId: preset?.partner?.id,
       volumes: AppAddController.buildVolumesBody(volumes),
       unsecureHttp: APP_PRIVACY_SETTINGS.PUBLIC === privacy,
-      scalingStrategy: {
-        fixed: {
-          replicas,
-        },
-      },
+      scalingStrategy: AppAddController.buildScalingeBody(scalingStrategy),
```
Comment (Dimfacion) [inline packages/manager/modules/pci/src/projects/project/ai/apps/add/add.controller.js:131]:
```suggestion
      scalingStrategy: AppAddController.buildScalingBody(scalingStrategy),
```

**Review by Dimfacion — APPROVED** [review]
Approving despite my nitpicking ;)

File: packages/manager/modules/pci/src/projects/project/ai/apps/add/add.controller.js (hunk 5 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/ai/apps/add/add.controller.js]: Changed here https://github.com/ovh/manager/pull/8001/commits/708f160c74073e7da19376d0277cbc6fde69e61a

File: packages/manager/modules/pci/src/projects/project/ai/apps/add/add.controller.js (hunk 6 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/ai/apps/add/add.controller.js]: Changed here: https://github.com/ovh/manager/pull/8001/commits/708f160c74073e7da19376d0277cbc6fde69e61a

**Pull request merged**